pkg/bu_server/broker/broker_test.go

```diff
@@ -206,3 +289,27 @@ func loadTradeDocument(datafile string) storage.TradeDocument {
 		Meta:         meta,
 	}
 }
+
+func loadPrivateKey(datafile string) any {
+	_, file, _, _ := runtime.Caller(1)
+	content, err := os.ReadFile(filepath.Clean(filepath.Join(filepath.Dir(file), datafile)))
+	if err != nil {
+		panic(err)
+	}
+	block, _ := pem.Decode(content)
+	if block == nil {
+		panic("failed to decode private key")
+	}
+	ecdsaKey, ecdsaErr := x509.ParseECPrivateKey(block.Bytes)
+	rsaKey, rsaErr := x509.ParsePKCS8PrivateKey(block.Bytes)
+	if ecdsaErr != nil && rsaErr != nil {
+		panic("failed to parse private key")
+	}
+	if ecdsaKey != nil {
+		return ecdsaKey
+	}
+	if rsaKey != nil {
+		return rsaKey
+	}
+	panic("no private key parsed")
```

Comment:
pkg/pkix/cert.go ParsePrivateKey can help you to read private key from PEM file.

Reply:
Updated.